docs/versioning.md

```diff
@@ -0,0 +1,35 @@
+# Versioning and Branching in AWS Load Balancer Operator
+
+The AWS Load Balancer Operator follows the semantic versioning, for any given release `X.Y.Z`:
+* an X (major) release indicates a set of backwards-compatible changes. Changing X means there's a breaking change.
+* a Y (minor) release indicates a minimum feature set. Changing Y means the addition of a backwards-compatible feature.
+* a Z (patch) release indicates minimum set of bugfixes. Changing Z means a backwards-compatible change that doesn't add functionality.
+
+## Branches
+
+The AWS Load Balancer Operator repository contains two types of branches: the `main` branch and `release-X.Y` branches.
+
+The main branch is where development happens. All the latest code, including breaking changes, happens on `main`.
+The `release-X.Y` branches contain stable, backwards compatible code. Every minor (`X.Y`) release, a new such branch is created.
+
+## Channels
+
+The AWS Load Balancer Operator's releases get published in two types of [OLM channels](https://olm.operatorframework.io/docs/glossary/#channel): the minor `release-vX.Y` and the major `release-vX`.
+
+The minor channels contain patch releases. The major channels contain all patch releases from all minor channels.
```

Comment:
I'm doubt if we should keep the major channels? I cannot see its use case, it that for upgrade automatically?

Reply:
> is that for upgrade automatically?

Yes. The idea was to simplify the upgrade and to minimize the configuration on the client side. Since we often publish new minor channels (practically every new feature) the client has to change its subscription to explicitly mention new channel. Also, the OCP docs don't have to updated every time we publish a new feature.